src/mixer.c

```diff
@@ -549,13 +549,27 @@ void mixTable(void)
         for (i = 0; i < numberMotor; i++)
             motor[i] = rcCommand[THROTTLE] * currentMixer[i].throttle + axisPID[PITCH] * currentMixer[i].pitch + axisPID[ROLL] * currentMixer[i].roll + -cfg.yaw_direction * axisPID[YAW] * currentMixer[i].yaw;
 
-    if (f.FIXED_WING) {
-        if (!f.ARMED)
-            motor[0] = mcfg.mincommand; // Kill throttle when disarmed
-        else
-            motor[0] = constrain(rcCommand[THROTTLE], mcfg.minthrottle, mcfg.maxthrottle);
+    if (f.FIXED_WING) { // vector_trust handeling
+        if (cfg.fw_vector_trust && f.ARMED) {
+            if (f.PASSTHRU_MODE) {
+                motor[0] = rcCommand[THROTTLE] - rcCommand[YAW] * 0.5f;
+                motor[1] = rcCommand[THROTTLE] + rcCommand[YAW] * 0.5f;
+            } else {
+                motor[0] = rcCommand[THROTTLE] - axisPID[YAW] * 0.5f;
+                motor[1] = rcCommand[THROTTLE] + axisPID[YAW] * 0.5f;
+            }
+        }
+        if (!cfg.fw_vector_trust) {
+            motor[0] = rcCommand[THROTTLE];
+            motor[1] = rcCommand[THROTTLE];
+        }
+        if (!f.ARMED || ((rcData[THROTTLE]) < mcfg.mincheck && feature(FEATURE_MOTOR_STOP))) {
+            motor[0] = mcfg.mincommand;
+            motor[1] = mcfg.mincommand;
+        }
     }

```

Comment:
There has to be a better way to write the above, I don't understand what current version does so surely there's a simpler way.